atlp-rwanda/code-ninjas-be
# #181414756 (feature): Send a verification email on registration

### What does this PR do?
 Verifies a user by their email
### Description of Task to be completed?
- Send a verification email with a token expiring after 5 minutes
- Update a user as verified 

### How should this be manually tested?
Get started By Running commands bellow:
  - clone repo
  - switch to ft-verify-email-181414756 branch
  - ``yarn install``: to install all dependencies
  - Create .env file
  - add environment variables as per .env.example

Run the test suite:
  - ``yarn test``: to run the test suite

Run the application locally:
- ``yarn dev``: to start the application in development mode
- open the application in your browser(you will be redirected to the documentation) and try these requests:
  - **/api/auth/register** (automatically sends a verification email as well)
  - **/api/users/send/confirm/{email}** (used to resend a verification email)
  - **/api/users/verify/{token}**
### Any background context you want to provide?
- Service API for sending emails of choice is Google
- Uses a redis stack to cache valid tokens 
### What are the relevant pivotal tracker stories?
[#181414756](https://www.pivotaltracker.com/story/show/181414756)
### Screenshots (if appropriate)
### Questions:
